File: backend_spring/src/main/java/com/buddy/pium/entity/common/Member.java
package com.buddy.pium.entity.common;

import com.buddy.pium.entity.chat.ChatRoomBan;
import com.buddy.pium.entity.chat.ChatRoomMember;
import com.buddy.pium.entity.chat.Message;
import com.buddy.pium.entity.calender.Calender;
import com.buddy.pium.entity.diary.Diary;
import com.buddy.pium.entity.post.Post;
import com.buddy.pium.entity.post.PostComment;
import com.buddy.pium.entity.post.PostLike;
import com.buddy.pium.entity.share.Share;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EntityListeners(AuditingEntityListener.class)
public class Member {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 100)
    private String email;

    @Column(nullable = false, length = 100)
    private String password;

    @Column(nullable = false, length = 50)
    private String username;

    @Column(nullable = false, unique = true, length = 50)
    private String nickname;

    @Column(length = 15)
    private String phoneNumber;

    @Column(length = 100)
    private String address;

    @Column
    private LocalDate birth;

    // M, F
    @Enumerated(EnumType.STRING)
    @Column(length = 1)
    private Enum.Gender gender;

    private String profileImageUrl;

    @Column
    private Long mateInfo;

    @Column(length = 255)
    private String refreshToken;

    @Builder.Default
    @OneToMany(mappedBy = "member", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ChatRoomMember> chatRoomMembers = new ArrayList<>();

    @Builder.Default
    @OneToMany(mappedBy = "bannedMember", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ChatRoomBan> chatRoomBan = new ArrayList<>();

    @Builder.Default
    @OneToMany(mappedBy = "sender", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Message> messages = new ArrayList<>();

    @Builder.Default
    @OneToMany(mappedBy = "member", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Child> children = new ArrayList<>();

    @Builder.Default
    @OneToMany(mappedBy = "sender", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<MateRequest> memberRequests = new ArrayList<>();

    @Builder.Default
    @OneToMany(mappedBy = "receiver", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<MateRequest> memberResponses = new ArrayList<>();

    @Builder.Default
    @OneToMany(mappedBy = "member", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Calender> calenders = new ArrayList<>();

    @Builder.Default
    @OneToMany(mappedBy = "member", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Diary> diaries = new ArrayList<>();

    @Builder.Default
    @OneToMany(mappedBy = "member", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Post> posts = new ArrayList<>();

    @Builder.Default
    @OneToMany(mappedBy = "member", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<PostComment> postComments = new ArrayList<>();

    @Builder.Default
    @OneToMany(mappedBy = "member", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<PostLike> likes = new ArrayList<>();

    @Builder.Default
    @OneToMany(mappedBy = "member", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Share> shares = new ArrayList<>();

    @CreatedDate
    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}

File: backend_spring/src/main/frontend_flutter/lib/models/member.dart
class Member {
  final int? id;
  final String? username;
  final String? nickname;
  final String? email;
  final String? password;
  final String? phoneNumber;
  final String? address;
  final String? birth; // LocalDate를 String으로 처리
  final String? gender; // Enum.Gender를 String으로 처리 ('M', 'F')
  final String? profileImage;
  final int? mateInfo;
  final String? createdAt; // Timestamp를 String으로 처리
  final String? updatedAt; // Timestamp를 String으로 처리

  Member({
    this.id,
    this.username,
    this.nickname,
    this.email,
    this.password,
    this.phoneNumber,
    this.address,
    this.birth,
    this.gender,
    this.profileImage,
    this.mateInfo,
    this.createdAt,
    this.updatedAt,
  });

  factory Member.fromJson(Map<String, dynamic> json) {
    return Member(
      id: json['id'],
      username: json['username'],
      nickname: json['nickname'],
      email: json['email'],
      password: json['password'],
      phoneNumber: json['phoneNumber'],
      address: json['address'],
      birth: json['birth'], // String으로 받음
      gender: json['gender'], // String으로 받음
      profileImage: json['profileImage'],
      mateInfo: json['mateInfo'],
      createdAt: json['createdAt'], // String으로 받음
      updatedAt: json['updatedAt'], // String으로 받음
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'username': username,
      'nickname': nickname,
      'email': email,
      'password': password,
      'phoneNumber': phoneNumber,
      'address': address,
      'birth': birth,
      'gender': gender,
      'profileImage': profileImage,
      'mateInfo': mateInfo,
      'createdAt': createdAt,
      'updatedAt': updatedAt,
    };
  }
}